ui_advanced_config.py

```diff
@@ -284,325 +401,219 @@ def BuildUI(self):
             padx=10,
             pady=10,
         )
+        self.mapsCreateButton.grid(
+            row=10,
+            column=1,
+            columnspan=2,
+            sticky="nsew",
+            ipadx=2,
+            ipady=2,
+            padx=10,
+            pady=10,
+        )
+        self.mapsDeleteButton.grid(
+            row=11,
+            column=1,
+            columnspan=2,
+            sticky="nsew",
+            ipadx=2,
+            ipady=2,
+            padx=10,
+            pady=10,
+        )
 
         self.initEntryWidgets()
 
     def initEntryWidgets(self):
-
+        # TEAM
         self.teamData = self.ldr.team_templates
         self.teamNames = self.ldr.getTeamNames()
+        print(self.teamNames)
+        self.currentTeamData = self.teamData[self.teamNames[0]]
+
         self.selectTeamCombo.configure(values=self.teamNames)
         self.selectTeamCombo.current(0)
-        self.selectTeamCombo.bind("<<ComboboxSelected>>", self.updateTeamEntryWidgets)
-        self.teamNameEntry.insert(0, self.teamNames[0])
-        self.teamSizeEntry.insert(0, self.teamData[self.teamNames[0]]["size"])
-        self.callsignEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["callsign"]
-        )
-        self.squadEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["squad"]
-        )
-        self.agentObjectEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["object"]
-        )
-        self.aiFileEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["AI_file"]
-        )
+        self.selectTeamCombo.bind("<<ComboboxSelected>>", self.changeTeamEntryWidgets)
+        self.showTeamEntry(self.currentTeamData)
 
+        # COMPONENT
         self.componentData = self.ldr.comp_templates
         self.componentIDs = self.ldr.getCompIDs()
         self.componentTypes = self.ldr.getCompTypes()
-        self.componentTypeAttr = self.componentData[self.componentIDs[0]].view_keys
+        self.currentComponentData = self.componentData[self.componentIDs[0]]
+        self.componentTypeAttr = self.currentComponentData.view_keys
+
         self.selectComponentCombo.configure(values=self.componentIDs)
         self.selectComponentCombo.current(0)
         self.selectComponentCombo.bind(
-            "<<ComboboxSelected>>", self.updateComponentsEntryWidgets
-        )
-        self.componentsIDEntry.insert(0, self.componentIDs[0])
-        self.componentsNameEntry.insert(
-            0, self.componentData[self.componentIDs[0]].getData("name")
-        )
-        self.componentsTypeCombo.configure(values=self.componentTypes)
-        self.componentsTypeCombo.set(
-            self.componentData[self.componentIDs[0]].getData("ctype")
+            "<<ComboboxSelected>>", self.changeComponentsEntryWidgets
```

Comment:
I don't think this is related to the new component, but is instead resulting from tkinterface trying to insert a null value into an entry widget. Certain component types like radio, have an attribute that can be null (for example for the radio, the message attribute can be null). Off the top of my head not sure the best way to fix this because in theory I think we might want people to edit this attribute, but then we would have to change it to a string that says "None" or something so it doesn't get mad.

Reply:
I believe these issues were resolved in my last commit.